src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala

```diff
@@ -40,19 +73,88 @@ class DataEntrySpecification extends PropSpec with PropertyChecks with Generator
+    }
+
+    forAll(validTimestampEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validBooleanEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validShortBytesEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validOpcBlockEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
+    forAll(validBigIntegerEntryGen) { pk: DataEntry =>
+      val recovered = DataEntry.fromBytes(pk.bytes).right.get
+
+      assertEys(recovered, pk)
+    }
+
   }
 
-  property("report invalid data type") {
+  property("report invalid and valid data type") {
     val byteArray1 = Array[Byte](0, 118, 97, 108, 117, 101, 49)
     val byteArray2 = Array[Byte](4, 118, 97, 108, 117, 101, 49)
     val byteArray3 = Array.fill[Byte](1 + 32)(1)
     val byteArray4 = Array.fill[Byte](1 + 26)(2)
     val byteArray5 = Array.fill[Byte](1 + 8)(3)
+    val byteArray6 = Array.fill[Byte](1 + 4)(4)
+    val byteArray7 = Array[Byte](5, 0, 0)
+    val byteArray8 = Array.fill[Byte](1 + 26)(6)
+    val byteArray9 = Array.fill[Byte](1 + 26)(7)
+    val byteArray10 = Array.fill[Byte](1 + 8)(9)
+    val byteArray11 = Array[Byte](10, 1)
+    val byteArray12 = Array[Byte](11, 0, 0)
+    val byteArray13 = Array[Byte](12, 118, 97, 108, 117, 101, 49)
+    val byteArray14 = Array[Byte](13, 0, 0)
+    val byteArray15 = Array[Byte](14, 0, 0)
     DataEntry.fromBytes(byteArray1) should be (Left(InvalidDataEntry))
     DataEntry.fromBytes(byteArray2) should be (Left(InvalidDataEntry))
     DataEntry.fromBytes(byteArray3).map(_.dataType) should be (Right(DataType.PublicKey))
     DataEntry.fromBytes(byteArray4).map(_.dataType) should be (Left(InvalidDataEntry))
     DataEntry.fromBytes(byteArray5).map(_.dataType) should be (Right(DataType.Amount))
+    DataEntry.fromBytes(byteArray6).map(_.dataType) should be (Right(DataType.Int32))
+    DataEntry.fromBytes(byteArray7).map(_.dataType) should be (Right(DataType.ShortText))
+    DataEntry.fromBytes(byteArray8).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray9).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray10).map(_.dataType) should be (Right(DataType.Timestamp))
+    DataEntry.fromBytes(byteArray11).map(_.dataType) should be (Right(DataType.Boolean))
+    DataEntry.fromBytes(byteArray12).map(_.dataType) should be (Right(DataType.ShortBytes))
+    DataEntry.fromBytes(byteArray13).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray14).map(_.dataType) should be (Right(DataType.OpcBlock))
+    DataEntry.fromBytes(byteArray15).map(_.dataType) should be (Right(DataType.BigInteger))
```

Comment:
@Icermli empty byte array is special here, may add more general cases

Reply:
Sure